cnd/src/lib.rs

```diff
@@ -81,15 +81,19 @@ pub fn data_dir() -> Option<PathBuf> {
     ProjectDirs::from("", "", "comit").map(|proj_dirs| proj_dirs.data_dir().to_path_buf())
 }
 
+#[cfg(target_os = "macos")]
+static LND_DIR_STEM: &str = "Lnd";
+#[cfg(target_os = "linux")]
+static LND_DIR_STEM: &str = ".lnd";
+#[cfg(target_os = "windows")]
+static LND_DIR_STEM: &str = "Lnd";
+
 /// Returns the LND data directory: $LND_DIR if it is set or else "~/.lnd".
-pub fn lnd_data_dir() -> PathBuf {
+pub fn lnd_dir() -> Option<PathBuf> {
     match env::var("LND_DIR") {
-        Ok(dir) => PathBuf::from(dir),
-        Err(_) => {
-            // Just unwrap, if we don't have $HOME we are hosed anyway.
-            let home = dirs::home_dir().unwrap();
-            home.join(".lnd")
-        }
+        Ok(dir) => Some(PathBuf::from(dir)),
+        Err(_) => ProjectDirs::from("", "", LND_DIR_STEM)
```

Comment:
Doesn't this evaluate to `$XDG_DATA_HOME/.lnd` or `$HOME/.local/share/.lnd` for Linux? According to the [docs](https://api.lightning.community/#lnd-grpc-api-reference), it should be `$HOME/.lnd` for Linux.

It looks like we'll need to do some messy conditional compiliation here ...
I would also recommend to add tests (that also have `cfg` applied) to make sure, we are actually resolving to the folders stated in the documentation.

Reply:
Man, some morning.  You are correct, this code is totally broken.

Reply:
So, the confusion is; on OSX lnd dir _is_ in the `$XDG_DATA_HOME` directory and on linux it is not.  Will fix.

Reply:
Yeah it is a mess unfortunately :/
But we can't really do anything about that (except maybe sending a PR to lnd :D)